_posts/2020-02-03-fetch-offset-ties.md

```diff
@@ -0,0 +1,99 @@
+---
+layout: post
+title:  "FETCH FIRST, OFFSET and TIES"
+author: Piotr Findeisen, Starburst Data
+excerpt_separator: <!--more-->
+---
+
+Presto follows the SQL Standard faithfully. We extend it only when it is well justified,
+we strive to never break it and we always prefer the standard way of doing things.
+There was one situation where we stumbled, though. We had a non-standard way of limiting
+query results with `LIMIT n` without implementing the standard way of doing that first.
+We have corrected that, adding ANSI SQL way of limiting query results, discarding initial
+results and -- a hidden gem -- retaining initial results in case of ties.
+
+<!--more-->
+
+# Limiting query results
+
+Probably everyone using relational databases knows the `LIMIT n` syntax for limiting query
+results. It is supported by e.g. MySQL, PostgreSQL and many more SQL engines following
+their example. It is so common that one could think that `LIMIT n` is the standard way
+of limiting the query results.  Let's have a look at how various popular SQL engines
+provide this feature.
+
+* DB2, MySQL, MariaDB, PostgreSQL, Redshift, MemSQL, SQLite and many others provide the `... LIMIT n` syntax.
+* SQL Server provides `SELECT TOP n ...` syntax.
+* Oracle provides `... WHERE ROWNUM <= n` syntax.
+
+And what does the SQL Standard say?
+
+```sql
+SELECT *
+FROM my_table
+FETCH FIRST n ROWS ONLY 
+```
+
+If we look again at the database systems mentioned above, it turns out many of them support the standard
+syntax too: Oracle, DB2, SQL Server and PostgreSQL (although that's not documented currently).
+
+And Presto? Presto has `LIMIT n` support since 2012. In [Presto 310](https://prestosql.io/docs/current/release/release-310.html),
```

Comment:
`310` instead of `Presto 310`?

Reply:
IMO "310" is not sufficient for describing link target.

Reply:
This renders as:
`And Presto? Presto has LIMIT n support since 2012. In Presto 310, Presto gained also the FETCH FIRST n ROWS ONLY support.`
Too much "Presto".
Maybe "Since 310, Presto gained..."

Reply:
Right. "In Presto 310, we added ..."